homeassistant/components/tradfri/sensor.py

```diff
@@ -23,84 +20,33 @@
     devices = (
         dev
         for dev in all_devices
-        if not dev.has_light_control and not dev.has_socket_control
+        if not dev.has_light_control
+        and not dev.has_socket_control
+        and not dev.has_blind_control
     )
-    async_add_entities(TradfriDevice(device, api) for device in devices)
+    if devices:
+        async_add_entities(TradfriSensor(device, api, gateway_id) for device in devices)
 
 
-class TradfriDevice(Entity):
+class TradfriSensor(TradfriBaseDevice):
     """The platform class required by Home Assistant."""
 
-    def __init__(self, device, api):
+    def __init__(self, device, api, gateway_id):
         """Initialize the device."""
-        self._api = api
-        self._device = None
-        self._name = None
-
-        self._refresh(device)
-
-    async def async_added_to_hass(self):
-        """Start thread when added to hass."""
-        self._async_start_observe()
-
-    @property
-    def should_poll(self):
-        """No polling needed for tradfri."""
-        return False
-
-    @property
-    def name(self):
-        """Return the display name of this device."""
-        return self._name
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit_of_measurement of the device."""
-        return "%"
+        super().__init__(device, api, gateway_id)
+        self._unique_id = f"{gateway_id}-{device.id}"
 
     @property
     def device_state_attributes(self):
         """Return the devices' state attributes."""
-        info = self._device.device_info
-        attrs = {
-            "manufacturer": info.manufacturer,
-            "model_number": info.model_number,
-            "serial": info.serial,
-            "firmware_version": info.firmware_version,
-            "power_source": info.power_source_str,
-            "battery_level": info.battery_level,
-        }
-        return attrs
+        return {ATTR_BATTERY_LEVEL: self._device.device_info.battery_level}
```

Comment:
If this is the same as state, we shouldn't return the value as a state attribute too.

Battery level sensors should preferably implement the `device_class` property and return `"battery"`.

https://developers.home-assistant.io/docs/en/entity_sensor.html#available-device-classes

Reply:
Updated